openvdb_houdini/houdini/SOP_OpenVDB_Scatter.cc

```diff
@@ -447,13 +505,17 @@ SOP_OpenVDB_Scatter::cookMySop(OP_Context& context)
 
             // add ALL the points to this group
             ptgroup->addRange(gdp->getPointRange());
-        }
 
-        // add the VDBs to the output if requested
-        if (1 == evalInt("keep", 0, time)) {
-            gdp->mergePrimitives(*vdbgeo, vdbgeo->getPrimitiveRange());
+            const std::string groupName(scatterStr.toStdString());
+            for (auto& pointGrid : pointGrids) {
+                openvdb::points::appendGroup(pointGrid->tree(), groupName);
+                openvdb::points::setGroup(pointGrid->tree(), groupName);
+            }
         }
 
+        for (auto& pointGrid : pointGrids) {
+            hvdb::createVdbPrimitive(*gdp, pointGrid, "points");
```

Comment:
I'd recommend a "VDB Name" field that appears when scattering VDB Points so as not to force the name "points". (As in OpenVDB Points Convert SOP).

Reply:
It could also be worth considering matching the name of the input VDB too...

Reply:
I'll definitely add a name field